File: examples/ecto/lib/shop/wishlist.ex
defmodule Shop.Wishlist do
  use Ecto.Schema
  @moduledoc """
  A saved list, one per account, holding products through a named implicit relation: the join
  table is `_Wished`, not `_ProductToWishlist`. Its key is a uuid and its owner's is an integer.
  """

  @primary_key {:id, :binary_id, autogenerate: true}
  @foreign_key_type :binary_id

  @type t :: %__MODULE__{
          id: Ecto.UUID.t(),
          share_token: String.t(),
          account: Shop.Account.t() | nil,
          products: [Shop.Product.t()]
        }

  schema "Wishlist" do
    field(:share_token, :string, autogenerate: {Ecto.UUID, :generate, []})
    belongs_to(:account, Shop.Account, foreign_key: :account_id, type: :id)
    many_to_many(:products, Shop.Product, join_through: "_Wished", join_keys: [B: :id, A: :id])
  end
end

File: examples/eloquent/app/Models/Wishlist.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasVersion4Uuids;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;

/**
 * A saved list, one per account, holding products through a named implicit relation: the join
 * table is `_Wished`, not `_ProductToWishlist`.
 */
class Wishlist extends Model
{
    use HasVersion4Uuids;

    protected $table = 'Wishlist';

    protected $keyType = 'string';

    public $incrementing = false;

    public $timestamps = false;

    protected $fillable = [
        'account_id',
    ];

    protected $casts = [
        'account_id' => 'integer',
    ];

    public function account(): BelongsTo
    {
        return $this->belongsTo(Account::class, 'account_id');
    }

    public function products(): BelongsToMany
    {
        return $this->belongsToMany(Product::class, '_Wished', 'B', 'A');
    }
}